### PR TITLE
Reactstrap: Added missing 'to' type definition to reactstrap navlink interface

### PR DESCRIPTION
**Pull request by janipiippow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/reactstrap/reactstrap/issues/298
This typing (to field in NavLink) didn't exist in the typings.
- [x] Increase the version number in the header if appropriate.
Not appropriate
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.
Not appropriate

### CODE REVIEW

File: types/reactstrap/lib/NavLink.d.ts
Comment (FaithForHumans) [inline types/reactstrap/lib/NavLink.d.ts]: I really hate to do this, but this isn't the correct way to do this as this isn't a reactstrap prop. See related discussion on #25329 that never got resolved.

I mentioned this in another comment chain on a different PR, but I've been meaning to put together a PR following the 

    export type RowProps<T> =  React.HTMLProps<HTMLElement> & T & {
        ...
    }
    declare class Row<T extends object = {}> extends React.Component<RowProps<T>> {}
    export default Row;

style syntax for all of the components, I just haven't had enough free time recently. Once #25921 gets merged, I'll try to put together a half hour or so to get that done since it'll cause a merge conflict.

**Review by Sprogel — CHANGES_REQUESTED** [review]

File: types/reactstrap/lib/NavLink.d.ts
Comment (FaithForHumans) [inline types/reactstrap/lib/NavLink.d.ts]: @deepu105 @timc13 

This has come up again. Thoughts?

Comment (timc13) [reply types/reactstrap/lib/NavLink.d.ts]: I am still against this and for generic components

Comment (deepu105) [reply types/reactstrap/lib/NavLink.d.ts]: @FaithForHumans @timc13, unfortunately, I don't have enough time to keep up with this, so I give up. As maintainers, I assume you guys know what to do best here and hope that you will make the typings useful for everyone. For time being I have decided not use this typing

Comment (janipiippow) [reply types/reactstrap/lib/NavLink.d.ts]: So what am I supposed to do? All I know, that there is a "to" field in the NavLink component that wasn't found in the typings, and I needed this field in my project. I did the obvious and added the field and thought that hey, this could be useful to others as well, as seen in https://github.com/reactstrap/reactstrap/issues/298, so why not share it. That wasn't good enough and if the fix that was pointed to me wasn't good enough, then what is? It works with both solutions. Maybe sometimes purism is not the best way to handle things, but to deliver value and useful content to the user.

Comment (FaithForHumans) [reply types/reactstrap/lib/NavLink.d.ts]: If you look at the reactstrap source, [it doesn't care what `to` is](https://github.com/reactstrap/reactstrap/blob/master/src/NavLink.js#L6) it's not one of the props that a NavLink cares about. [Reactstrap passes along all props it doesn't care about](https://github.com/reactstrap/reactstrap/blob/master/src/NavLink.js#L50). Someone else might need `to` to be a boolean instead of a string if they're using something other than `react-router` for example.

Since reactstrap does pass along the other props, there needs to be a way to capture this fact as evidenced by this PR. There's two primary ways to do this: `[key: string]: any` or the generics method (what was in my previous comment above). That's what the discussion in #25329 was about, which is "better". Deepu105 pushed for the `[key: string]: any` method, and timc13 pushed for the generics method. You can read more detail in that PR.

I said "This has come up again" since it never really got resolved last time, and I didn't want to sneak through a PR without bringing it to their attention. Please don't misunderstand my intentions; I really do appreciate more people willing to pitch in, and I appreciate you bringing this PR forward as it got the ball rolling again.

If you have preference and more points for the `[key: string]: any` method over generics, please, field them here. Personally, I can see it going either way. That being said, I'll have time tomorrow to put together a PR to make all of the components generics that will fix this issue if we want to go down that path instead.

Comment (FaithForHumans) [reply types/reactstrap/lib/NavLink.d.ts]: As promised, #26094 contains the generics replacement.

**Comment by FaithForHumans** [comment]
@janipiippow Since #26094 fixes this and has been merged, are you good with closing this?

**Comment by janipiippow** [comment]
@FaithForHumans Sure, let's close it

**Pull request closed without merging**
